presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -191,7 +192,17 @@ public void destroy()
     public void start()
     {
         if (started.compareAndSet(false, true)) {
-            refreshExecutor.scheduleWithFixedDelay(this::refreshAndStartQueries, 1, 1, TimeUnit.MILLISECONDS);
+            refreshExecutor.scheduleWithFixedDelay(() -> {
+                try {
+                    refreshAndStartQueries();
+                }
+                catch (Throwable t) {
+                    log.error(t, "Error while executing refreshAndStartQueries");
+                    throw t;
```

Comment:
Let's suppress this Exception per the docs: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay-java.lang.Runnable-long-long-java.util.concurrent.TimeUnit-

Logging should be sufficient.

Reply:
OR, what we could do is stash the returned `ScheduledFuture` from `refreshExecutor.scheduleWithFixedDelay`, and use it to see if this as completed or failed (failing queries if so).

Reply:
Are all queries stuck or only some queries? If some queries are still running, it may be other reasons.
Logging is enough, just like other places we handle error from `scheduleWithFixedDelay`. You can throw error in custom build for verification purpose.